### PR TITLE
More on auto-link

### PR DESCRIPTION
**Pull request by BurdetteLamar**
@nobu [suggests](https://github.com/ruby/ruby/pull/9842#discussion_r1481348817) that a reference to a class such as `IO` (even the local one) should be `+IO+` (to force monofont), not `\IO` (plain text).  In my view, if it's not plain text, it may as well be auto-linked (`IO`), since both forced monofont and auto-linking cause both the font and the background color to be changed.

Therefore I propose allowing auto-linking in almost all cases.

This proposed change does not imply that current docs should be revised for this purpose, but only that additions or revisions should consider this.

### CODE REVIEW

File: doc/contributing/documentation_guide.md
Comment (jeremyevans) [inline doc/contributing/documentation_guide.md:181]: I recommend removing the last bullet as well.  It mentions `Class`, `Module`, and `Method`.  First, I'm not sure users don't usually deal with them.  Second, even if they don't usually deal with them, there is no problem auto-linking when referring to the Ruby class/entity.

I think the only time auto-linking should be supressed is if the usage does not refer to the Ruby class/entity.  Maybe:

```
In general, \RDoc's auto-linking should not be suppressed.
For example, we should write `Array`, not `\Array`.
However, suppress when the word in question does not refer to the Ruby entity
(e.g., some uses of _Class_ or _English_).
```

**Review by jeremyevans — CHANGES_REQUESTED** [review]

File: doc/contributing/documentation_guide.md
Comment (BurdetteLamar) [reply doc/contributing/documentation_guide.md]: Done.

**Review by jeremyevans — APPROVED** [review]
You may want to remove the blank line above the new paragraph, but that's up to you.

**Comment by peterzhu2118** [comment]
> In my view, if it's not plain text, it may as well be auto-linked (IO), since both forced monofont and auto-linking cause both the font and the background color to be changed.

I don't agree. Using monofont signals that it's code, but linking to the same page (except linking to a particular section) is pointless and distracts the user from which links are useful to click on.

**Comment by BurdetteLamar** [comment]
> > In my view, if it's not plain text, it may as well be auto-linked (IO), since both forced monofont and auto-linking cause both the font and the background color to be changed.
> 
> I don't agree. Using monofont signals that it's code, but linking to the same page (except linking to a particular section) is pointless and distracts the user from which links are useful to click on.

You'd be comfortable with code for page self-reference, and auto-links for others?

File: doc/contributing/documentation_guide.md
Comment (nobu) [inline doc/contributing/documentation_guide.md:174]: No period at the end?

**Review by nobu — APPROVED** [review]

File: doc/contributing/documentation_guide.md
Comment (BurdetteLamar) [reply doc/contributing/documentation_guide.md]: Fixed.

**Comment by BurdetteLamar** [comment]
Once we have consensus/approval for the specific reviewers here, I'd like to leave this open for a few days for any others to comment.

**Review by jeremyevans — APPROVED** [review]

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**
